pkg/threescale/backend/manager.go

```diff
@@ -220,58 +220,85 @@ func (b *Backend) Flush() {
 		}
 
 		for _, cachedApp := range service {
+			if cachedApp.reportingErr && cachedApp.authErr {
+				// nothing to do here only continue on
+				// todo add logging
+				continue
+			}
+
+			//now that we have handled the dual error case, we know we need to
+			// read from the cache and modify the entry so we take a write lock
```

Comment:
Would it be possible at this point that the `cache.Get` below fails?

Reply:
you would expect not, but i added the check to ensure we hit and if we miss just skip. Id rather do that than risk a crash, wdyt?